src/mono/wasm/runtime/jiterpreter-support.ts

```diff
@@ -2033,3 +2094,43 @@ function updateOptions() {
         (<any>optionTable)[k] = blob[info];
     }
 }
+
+function jiterpreter_allocate_table(type: JiterpreterTable, base: number, size: number, fillValue: Function) {
+    const wasmTable = getWasmFunctionTable();
+    const firstIndex = base, lastIndex = firstIndex + size - 1;
+    mono_assert(lastIndex < wasmTable.length, () => `Last index out of range: ${lastIndex} >= ${wasmTable.length}`);
+    for (let i = firstIndex; i <= lastIndex; i++)
+        wasmTable.set(i, fillValue);
+    cwraps.mono_jiterp_initialize_table(type, firstIndex, lastIndex);
+    return base + size;
+}
+
+// a single js worker might end up hosting multiple managed threads over its lifetime.
+// we need to ensure we only ever initialize tables once on each js worker.
+let jiterpreter_tables_allocated = false;
+
+export function jiterpreter_allocate_tables(module: any) {
+    if (jiterpreter_tables_allocated)
+        return;
+    jiterpreter_tables_allocated = true;
+
+    const options = getOptions();
+    // FIXME: Unfortunately the interp entry tables need to be REALLY big. I'm not sure why.
+    // A partial solution would be to merge the tables based on argument count instead of exact type,
+    //  then create special placeholder functions that examine the rmethod to determine which kind
+    //  of method is being called.
+    const tableSize = options.tableSize, interpEntryTableSize = options.aotTableSize,
```

Comment:
If you somehow changed these values between startup and creation of the new thread, yes. Realistically speaking it's not possible for that to happen. The jiterpreter will notice that the values don't match and it will assert